File: src/main/java/service/datetime/impl/RandomDateCreator.java
package service.datetime.impl;

import java.sql.Date;
import java.time.LocalDateTime;
import java.time.Month;
import java.time.ZoneOffset;
import java.time.temporal.ChronoField;

import org.springframework.stereotype.Component;

import dto.db.ColumnDefinitionDto;
import service.RandomDataCreator;

@Component
public class RandomDateCreator implements RandomDataCreator<Date> {

	// 2018/05/10 00:00:00.000
	private LocalDateTime ldt = LocalDateTime.of(2018, Month.JULY, 30, 0, 0, 0, 0);

	@Override
	public Date create(ColumnDefinitionDto cellDef, int seed, int dataNo) {
		LocalDateTime retLdt = null;
		int dayOfYear = ldt.get(ChronoField.DAY_OF_YEAR);

		dayOfYear = dayOfYear + seed + dataNo;
		if (dayOfYear > 365) {
			int yearAdd = dayOfYear / 365;
			dayOfYear = dayOfYear % 365 + 1;
			retLdt = ldt.with(ChronoField.YEAR, ldt.get(ChronoField.YEAR) + yearAdd);
			retLdt = retLdt.with(ChronoField.DAY_OF_YEAR, dayOfYear);
		} else {
			retLdt = ldt.with(ChronoField.DAY_OF_YEAR, dayOfYear);
		}
		return new Date(retLdt.toInstant(ZoneOffset.ofHours(9)).getEpochSecond() * 1000);
	}
}

File: src/main/java/tdc/service/datetime/impl/RandomDateCreator.kt
package tdc.service.datetime.impl

import org.springframework.stereotype.Component
import tdc.db.ColumnDefinitionDto
import tdc.service.RandomDataCreator
import java.sql.Date
import java.time.LocalDateTime
import java.time.Month
import java.time.ZoneOffset
import java.time.temporal.ChronoField

@Component
class RandomDateCreator : RandomDataCreator<Date?> {
    // 2018/05/10 00:00:00.000
    private val ldt = LocalDateTime.of(2018, Month.JULY, 30, 0, 0, 0, 0)
    override fun create(cellDef: ColumnDefinitionDto?, seed: Int, dataNo: Int): Date {
        var retLdt: LocalDateTime? = null
        var dayOfYear = ldt[ChronoField.DAY_OF_YEAR]
        dayOfYear += seed + dataNo
        if (dayOfYear > 365) {
            val yearAdd = dayOfYear / 365
            dayOfYear = dayOfYear % 365 + 1
            retLdt = ldt.with(ChronoField.YEAR, (ldt[ChronoField.YEAR] + yearAdd).toLong())
            retLdt = retLdt.with(ChronoField.DAY_OF_YEAR, dayOfYear.toLong())
        } else {
            retLdt = ldt.with(ChronoField.DAY_OF_YEAR, dayOfYear.toLong())
        }
        return Date(retLdt.toInstant(ZoneOffset.ofHours(9)).epochSecond * 1000)
    }
}